onnx2kerastl/linear_layers.py

```diff
@@ -28,31 +28,33 @@ def convert_gemm(node, params, layers, lambda_func, node_name, keras_name):
         logger.debug('Convert GEMM without bias.')
     else:
         raise AttributeError('More than 3 or less than 2 inputs')
-
+    keras_input = layers[node.input[0]]
     # Linear can have additional flag to transpose weights
     if 'transB' in params and params['transB'] == 1:
         logger.debug('Transposing W matrix.')
         keras_weights[0] = keras_weights[0].transpose()
     # Estimate input/output neurons
     input_channels, output_channels = keras_weights[0].shape[-2:]
+    if not keras_input.shape[-1] == input_channels:
+        keras_input = tf.transpose(keras_input)
```

Comment:
assuming the onnx model we get is well functioning how do you explain that?

Reply:
there was a fix in our code that should do this exact transpose but it is implemented with try-except (lines 52-57). I tried to find a specific condition. I am not sure how the onnx model is handling that, this is a good question, I'll look into that